web/src/toga_web/widgets/timeinput.py

```diff
@@ -0,0 +1,80 @@
+import datetime
+
+from toga_web.libs import create_proxy
+
+from .base import Widget
+
+
+def py_time(native_time):
+    if isinstance(native_time, datetime.datetime):
+        return native_time.time().replace(microsecond=0)
+
+    if isinstance(native_time, str):
+        parsed_time = datetime.time.fromisoformat(native_time)
+        return parsed_time.replace(microsecond=0)
+
+    return datetime.datetime.now().time().replace(microsecond=0)
+
+
+def native_time(py_time):
+    if isinstance(py_time, datetime.datetime):
+        return py_time.time().replace(microsecond=0).isoformat()
+
+    return py_time.replace(microsecond=0).isoformat()
+
+
+class TimeInput(Widget):
+
+    def create(self):
+        self.native = self._create_native_widget("sl-input")
+        self.native.setAttribute("step", "1")  # force seconds to show
+        self.native.type = "time"
+        self.native.value = native_time(
+            datetime.datetime.now().time().replace(microsecond=0)
+        )
+        self.native.addEventListener("sl-blur", create_proxy(self.dom_sl_blur))
+        self.native.addEventListener("sl-change", create_proxy(self.dom_sl_change))
+
+    def dom_sl_change(self, event):
+        self.interface.on_change()
+
+    def dom_sl_blur(self, event):
```

Comment:
I've done some testing after DateInput landed, and it looks like `sl-blur` is redundant here - `sl-change` is the only event we care about. If the value changes, it needs to be normalised, the event propagated. We don't need to normalise when focus is lost.